File: Tema 06 - React/petshelter.ui/src/Models/Pet.ts
import { Person } from "./Person";

export enum AnimalType{
    Cat="Cat",
    Dog = "Dog"
}

export class Pet {
    id: number;
    name: string;
    imageUrl: string;
    type: AnimalType;
    description: string;
    birthdate: Date;
    isHealthy: boolean;
    weightInKg: number;
    isSheltered: boolean;
    rescuer: Person;
    adopter?: Person;

    constructor(id: number, name: string, imageUrl: string, type: AnimalType, description:string, birthdate: Date, isHealthy: boolean, weightInKg: number, isSheltered: boolean, rescuer: Person, adopter?: Person) {
        this.id = id;
        this.name = name;
        this.imageUrl=imageUrl;
        this.type = type;
        this.description = description;
        this.birthdate = birthdate;
        this.isHealthy = isHealthy;
        this.weightInKg = weightInKg;
        this.isSheltered = isSheltered;
        this.rescuer = rescuer;
        this.adopter = adopter;
    }
}

File: Tema 08 - SOLID & Design Patterns/PetShelter/PetShelter.DataAccessLayer/Models/Pet.cs
namespace PetShelter.DataAccessLayer.Models;

public class Pet: IEntity
{
    /// <summary>
    ///     Identifier
    /// </summary>
    public int Id { get; set; }

    public string Name { get; set; }

    public string ImageUrl { get; set; }

    /// <summary>
    ///     e.g. Cat, Dog etc.
    /// </summary>
    public string Type { get; set; }

    public string? Description { get; set; }

    public DateTime Birthdate { get; set; }
    
    public bool IsHealthy { get; set; }
    
    public decimal WeightInKg { get; set; }

    /// <summary>
    ///     Can be used for soft delete. The pet can be marked as not sheltered anymore if it was adopted or it died.
    /// </summary>
    public bool IsSheltered { get; set; }

    /// <summary>
    ///     FK to a person
    /// </summary>
    public int? RescuerId { get; set; }

    /// <summary>
    ///     FK to a person
    /// </summary>
    public int? AdopterId { get; set; }

    public Person? Adopter { get; set; }
    public Person Rescuer { get; set; }
}
